File: plant-front/src/models/transaction.ts
import {IPlant, Plant} from './plant';
import {IUser, User} from './user';
import {IPlantSize, PlantSize} from './plantSize';
import {ITransactionType, TransactionType} from './transactionType';
import {DeliveryOption, IDeliveryOption} from './deliveryOption';

export interface ITransaction {
  id?: number;
  sellerId?: number;
  seller?: IUser;
  buyerId?: number;
  buyer?: IUser;
  plantId?: number;
  plant?: IPlant;
  transactionTypes?: ITransactionType[];
  plantSizeId?: number;
  plantSize?: IPlantSize;
  description?: string;
  locationPostalCode?: string;
  locationCity?: string;
  wish?: string;
  price?: number;
  deliveryOptions?: IDeliveryOption[];
  start?: Date;
  end?: Date;
  isClosed?: boolean;
}

export class Transaction {
  Buyer?: User;
  BuyerId?: number;
  DeliveryOptions: DeliveryOption[];
  Description?: string;
  End?: Date;
  Id?: number;
  LocationCity?: string;
  LocationPostalCode?: string;
  Plant?: Plant;
  PlantId?: number;
  PlantSize?: PlantSize;
  PlantSizeId?: number;
  Price?: number;
  Seller?: User;
  SellerId?: number;
  Start?: Date;
  TransactionTypes: TransactionType[];
  IsClosed: boolean;

  constructor(input: ITransaction) {
    this.Buyer = input.buyer ? new User(input.buyer) : undefined;
    this.BuyerId = input.buyerId;
    this.DeliveryOptions = input.deliveryOptions ? input.deliveryOptions.map(d => new DeliveryOption(d)) : [];
    this.Description = input.description;
    this.End = input.end;
    this.Id = input.id;
    this.LocationCity = input.locationCity;
    this.LocationPostalCode = input.locationPostalCode;
    this.Plant = input.plant ? new Plant(input.plant) : undefined;
    this.PlantId = input.plantId;
    this.PlantSize = input.plantSize ? new PlantSize(input.plantSize) : undefined;
    this.PlantSizeId = input.plantSizeId;
    this.Price = input.price;
    this.Seller = input.seller ? new User(input.seller) : undefined;
    this.SellerId = input.sellerId;
    this.Start = input.start;
    this.TransactionTypes = input.transactionTypes ? input.transactionTypes.map(t => new TransactionType(t)): [];
    this.IsClosed = input.isClosed || false;
  }
}

File: plant-back/PlantLibrary/Models/Transaction.cs
using SharedLibrary.Database;
using System;
using System.Collections.Generic;

namespace PlantLibrary.Models
{
    [BsonCollection("Transaction")]
    public class Transaction : Document
    {
        public string Description { get; set; }
        public string LocationPostalCode { get; set; }
        public string LocationCity { get; set; }
        public double Price { get; set; } // TODO : if sale
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public bool IsClosed { get; set; } // Default : false
        public Plant Plant { get; set; }
        public PlantSize PlantSize { get; set; }
        public User Seller { get; set; }
        public User Buyer { get; set; }
        public List<DeliveryOption> DeliveryOptions { get; set; }
        public List<TransactionType> TransactionTypes { get; set; }
        public List<Picture> Pictures { get; set; }


        public List<TransactionProposal> Proposals { get; set; }


    }

}
